quantities/quantities.hpp

```diff
@@ -104,8 +104,8 @@ class Quantity final {
   template<typename Q>
   friend constexpr Q SIUnit();
 
-  template<typename D>
-  friend __m128d ToM128D(Quantity<D> x);
+  template<typename Dimensions>
```

Comment:
Why don't we call the template parameter of `Quantity` `Dimensions_`?  That's what we do these days to avoid name clashes.